extensions/amp-fresh/0.1/amp-fresh-manager.js

```diff
@@ -14,14 +14,99 @@
  * limitations under the License.
  */
 
+import {addParamToUrl} from '../../../src/url';
 import {fromClass} from '../../../src/service';
+import {isExperimentOn} from '../../../src/experiments';
+import {whenDocumentReady} from '../../../src/document-ready';
+import {user} from '../../../src/log';
+import {xhrFor} from '../../../src/xhr';
 
 
+/** @const */
+const TAG = 'amp-fresh';
+
 /**
  * Manages registered amp-fresh components and does the xhr request
  * and update.
  */
-export class AmpFreshManager {}
+export class AmpFreshManager {
+
+  /**
+   * @param {!Window} win
+   */
+  constructor(win) {
+
+    /** @type {!Window} */
+    this.win = win;
+
+    if (!isExperimentOn(this.win, TAG)) {
+      return;
+    }
+
+    /** @private @const {!Object<string, !./amp-fresh.AmpFresh>} */
+    this.ampFreshInstances_ = Object.create(null);
+
+    /** @private {?Document} */
+    this.docPromise_ = null;
+
+    this.fetchDocument_().then(doc => {
```

Comment:
I could imagine that we could delay body visibility for a little bit, just as we do with fonts. But I think my suggestion above to go w/o `amp-hidden` kind of does this already.